src/config/ConfigHelper.ts

```diff
@@ -338,8 +338,9 @@ export class ConfigHelper {
         config.sdk = 'evm'
       }
     }
-
-    config.accessListFactory = contractAddressesConfig.accessListFactory
+    if ('accessListFactory' in contractAddressesConfig && config.sdk === 'oasis') {
```

Comment:
is the `accessListFactory`  ONLY related with the oasis network?
I don't think so

Reply:
> is the accessListFactory ONLY related with the oasis network?
I don't think so. Its related with template 4, not a specific chain (even if for the moment is only oasis using it)
What are we exactly trying to achieve here?

It's related with same networks, for example on polygon you don't have accesslist factory. We are trying to achieve here is not to enforce accessListFactory and to check if it exists. Maybe the check with config.sdk shouldn't be there, because you can have accessListFactory on other networks

Reply:
Not sure if i made myself clear, but the current change does not solve the issue, its wrong
we should check if `contractAddressesConfig` exists..
and only then do this:
`config.accessListFactory = contractAddressesConfig.accessListFactory` (this is OK)
This error:
`Error publishing asset: Cannot read properties of undefined (reading 'accessListFactory')`
means `contractAddressesConfig` is `undefined` and we're trying to access the property `accessListFactory` from it 

the change should be something like:
```
if(contractAddressesConfig) {
  config.accessListFactory = contractAddressesConfig.accessListFactory
}
```